libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/LoginActivity.java

```diff
@@ -191,8 +190,8 @@ private void completeAuthFlow(Intent intent) {
             final String errorDesc = params.get("error_description");
             webviewHelper.onAuthFlowError(error, errorDesc, null);
         } else {
-            final OAuth2.TokenEndpointResponse tr = new OAuth2.TokenEndpointResponse(params);
-            webviewHelper.onAuthFlowComplete(tr);
+            String code = params.get("code");
```

Comment:
We expect to get a code then we give it to onWebServerFlowComplete which will exchange it for tokens.

Reply:
Here we are inside completeAuthFlow(Intent intent) which is called during advanced authentication.